lms/envs/common.py

```diff
@@ -5379,6 +5382,26 @@ def _make_locale_paths(settings):  # pylint: disable=missing-function-docstring
 NOTIFICATIONS_EXPIRY = 60
 EXPIRED_NOTIFICATIONS_DELETE_BATCH_SIZE = 10000
 
+######################## Event bus producer config ########################
+
+# .. setting_name: EVENT_BUS_PRODUCER_CONFIG
+# .. setting_default: {}
+# .. setting_description: Dictionary of event_types mapped to lists of dictionaries containing topic related configuration.
+#    Each topic configuration dictionary contains
+#    * a topic/stream name called `topic` where the event will be pushed to.
+#    * a flag called `enabled` denoting whether the event will be published to the topic.
+#    * `event_key_field` which is a period-delimited string path to event data field to use as event key.
+#    Note: The topic names should not include environment prefix as it will be dynamically added based on
+#    EVENT_BUS_TOPIC_PREFIX setting.
+EVENT_BUS_PRODUCER_CONFIG = {
+    'org.openedx.learning.certificate.created.v1': [
```

Comment:
Can we add toggle annotations for each of these following: https://edx.readthedocs.io/projects/edx-toggles/en/latest/how_to/documenting_new_feature_toggles.html?

Reply:
The best I can think of is "temporary" but we also don't have a removal date, which is required. opt_in, maybe? or open_edx?

Reply:
it seems dependent on https://github.com/openedx/openedx-events/issues/265

Reply:
Good questions.
1. Because the toggle is a permanent feature of the config, let's not call it temporary.
2. I think `opt_in` works fine. Note that these options should be going away anyway.
3. We have a strange situation where we will think we might be changing the default after some roll-out period, and this is dependent on the ticket you linked. Maybe just have some comment around this at the top of the config, noting that we need to work this out?

Reply:
Looking more I think we may still want to keep 'ENABLE_SEND_XBLOCK_EVENTS_OVER_BUS' since it actually gates several events which presumably should be launched or held back as a group. We can use the value of it to set the value in the config map. What do you think?

Reply:
Wait jk changing that flag in a separate configuration file would not actually change the value in the event bus config map since presumably the separate file is parsed afterwards. I guess I'll just put a toggle_warning that they should all be the same

Reply:
Some links of potential interest:
* https://github.com/openedx/edx-platform/blob/3c3306ce855d508ccae2d5b38607e47afe90db27/openedx/core/lib/derived.py#L14
* https://djangodeconstructed.com/2018/05/25/lazy-django-how-django-accesses-project-settings-setup-part-1/

Reply:
Are you concerned with a single topic, or the entire setting? We could also use [KEYS_WITH_MERGED_VALUES](https://github.com/openedx/edx-platform/blob/9b44065ab08aab7f992cd328ea09cbfcdb3a5149/lms/envs/production.py#L77C5-L77C28).

Reply:
The entire setting. If they wanted to change a single topic, they'd have to copy over the entire config dict and set all the derived settings correctly, which is non obvious. Keys with merged values looks like it may be our friend here.